adapters/Nend/NendAdapter/nativeads/GADMAdapterNendNativeVideoAd.m

```diff
@@ -0,0 +1,169 @@
+//
+//  GADMAdapterNendNativeVideoAd.m
+//  NendAdapter
+//
+//  Copyright © 2019 FAN Communications. All rights reserved.
+//
+
+#import "GADMAdapterNendNativeVideoAd.h"
+#import "GADNendNativeAdLoader.h"
+
+@interface GADMAdapterNendNativeVideoAd () <NADNativeVideoDelegate, NADNativeVideoViewDelegate>
+
+@property(nonatomic, strong) NADNativeVideo *videoAd;
+@property(nonatomic, strong) NADNativeVideoView *nendMediaView;
+@property(nonatomic, strong) GADNativeAdImage *mappedIcon;
+@property(nonatomic, strong) NSDecimalNumber *userRating;
+
+@end
+
+@implementation GADMAdapterNendNativeVideoAd
+
+- (instancetype)initWithVideo:(NADNativeVideo *)ad {
+    self = [super init];
+    if (self) {
+        _videoAd = ad;
+        _videoAd.delegate = self;
+
+        _nendMediaView = [NADNativeVideoView new];
+        _nendMediaView.delegate = self;
+
+        _mappedIcon = [[GADNativeAdImage alloc] initWithImage:ad.logoImage];
+        _userRating = [[NSDecimalNumber alloc] initWithFloat:ad.userRating];
+    }
+    return self;
+}
+
+- (BOOL)hasVideoContent {
+    return self.videoAd.hasVideo;
+}
+
+- (UIView *)mediaView {
+    return self.nendMediaView;
+}
+
+- (NSString *)advertiser {
+    return self.videoAd.advertiserName;
+}
+
+- (NSString *)headline {
+    return self.videoAd.title;
+}
+
+- (NSArray *)images {
+    return nil;
+}
+
+- (NSString *)body {
+    return self.videoAd.explanation;
+}
+
+- (GADNativeAdImage *)icon {
+    return self.mappedIcon;
+}
+
+- (NSString *)callToAction {
+    return self.videoAd.callToAction;
+}
+
+- (NSDecimalNumber *)starRating {
+    return self.userRating;
+}
+
+- (NSString *)store {
+    return nil;
+}
+
+- (NSString *)price {
+    return nil;
+}
+
+- (NSDictionary *)extraAssets {
+    return nil;
+}
+
+- (UIView *)adChoicesView {
+    return nil;
```

Comment:
Nend has no adchoices view? But they have a nadNativeVideoDidClickInformation callback. What view is that, and can it be propagated here?

Reply:
nend's native-video-ad provides AdChoises View on the `NendAdNativeMediaView`.

<img width="370" alt="スクリーンショット 2020-02-21 15 31 13" src="https://user-images.githubusercontent.com/26888328/75010054-89773880-54bf-11ea-86eb-dc6494b0b126.png">

And it's cannot separate from Video View.

Reply:
This is still nil!

Reply:
>This is still nil!

Because our AdChoises View cannot separate from `NendAdNativeMediaView`.
And it occurs to display multiple AdChoises View when mediated nend's native-videoads if return AdChoises View's object via `adChoicesView`.

Reply:
could you provide an unit id to test it  I can see the Native adds with small-square image, large-wide-image, and text only, but not videos.

Reply:
![Screen Shot 2020-03-03 at 1 33 00 PM](https://user-images.githubusercontent.com/59298836/75821724-9879a880-5d53-11ea-9173-5ca0c128044c.png)